File: src/main/java/com/twp/common/component/process/map/combo/IComboMapRouterProcessFunc.java
package com.twp.common.component.process.map.combo;

import com.twp.common.component.process.map.IVoidMapRouterProcessFunc;
import com.twp.common.component.process.statemachine.domain.iinterface.IMapRouterProcessStateNodeFunc;

import java.util.Map;

/**
 * Created by tanweiping on 2018/7/4.
 */
public interface IComboMapRouterProcessFunc {
     <T> T routerFunc(T t,Map<T, ? extends IMapRouterProcessStateNodeFunc<T>> routeFuncMap);

     <T> T routerFunc(T t,Map<T, ? extends IMapRouterProcessStateNodeFunc<T>> routeFuncMap,IExactFilter<T> matchFunc);

     <T> void routerVoidFunc(T t,Map<T, ? extends IVoidMapRouterProcessFunc> routeFuncMap);
     <T> void routerVoidFunc(T t,Map<T, ? extends IVoidMapRouterProcessFunc> routeFuncMap,IExactFilter<T> matchFunc);
}

File: src/main/java/com/twp/common/component/process/map/combo/IComboMapRouterProcessFunc.kt
package com.twp.common.component.process.map.combo

import com.twp.common.component.process.map.IVoidMapRouterProcessFunc
import com.twp.common.component.process.statemachine.domain.StateNode
import com.twp.common.component.process.statemachine.domain.iinterface.IMapRouterProcessStateNodeFunc
import com.twp.common.tuple.ResultDTO
import org.omg.CORBA.Object


interface IComboMapRouterProcessFunc {
    fun <T> routerFunc(t: T, routeFuncMap: Map<T, IMapRouterProcessStateNodeFunc<T>>): T

    fun <T> routerFunc(t: T, routeFuncMap: Map<T, IMapRouterProcessStateNodeFunc<T>>,matchFunc:IExactFilter<T>?): T

    fun <T> routerVoidFunc(t: T, routeFuncMap: Map<T, IVoidMapRouterProcessFunc>)
    fun <T> routerVoidFunc(t: T, routeFuncMap: Map<T, IVoidMapRouterProcessFunc>,matchFunc:IExactFilter<T>?)

}
